### PR TITLE
Add unit testing with Pytest and Coverage

### PR DESCRIPTION
**Pull request by Loic-Ker**
- You can run coverage run -m pytest tests/ to run the tests
- And then coverage report -m gives a the coverage report on the tests

### CODE REVIEW

**Comment by Loic-Ker** [comment]
You can maybe add something like this in tox.init to run pytest as a github action : 

commands =
  pytest -v
deps = 
  pytest

**Comment by peterampazzo** [comment]
I would wait to add CI/CD and pre-commit hooks for these tests since they're mostly testing the data. Even if pre-commits are run locally, it would take some time to run the checks at every commit done. Therefore, I think it would make sense to have few more tests for the current code/function (eg. testing the model architecture, by inspecting the class) which are more feasible to validate in short time using CI/CD.

**Review by peterampazzo — CHANGES_REQUESTED** [review]
test the current code and not only the data 😉

File: tests/not_test_data/test_transform.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import torch
+from src.models.conv_nn import ConvNet
+from src.models.kornia_trans import transform
+
+
+def test_transform():
+    data = torch.rand([64, 3, 224, 224]) # random datas
+    model = ConvNet(1028,512)
```
Comment (alessandro-canevaro) [inline tests/not_test_data/test_transform.py]: I think this line (8) can be removed

**Comment by peterampazzo** [comment]
I added the `tests/` folder to the checks in tox and fix some errors.

File: tests/not_test_data/test_transform.py (hunk 1 above)
Comment (peterampazzo) [reply tests/not_test_data/test_transform.py]: indeed, flake8 was raising this error. fixed with bb191ef.

**Review by peterampazzo — APPROVED** [review]

**Review by erlahrafnkels — APPROVED** [review]

**Review by alessandro-canevaro — APPROVED** [review]

**Pull request merged**